### ISSUE TITLE
Multi-GPU support in embeddings.py

### ISSUE THREAD

**Issue by staszekdh** [labstructbioinf/pLM-BLAST#27]
```
python embeddings.py start data_set_fullseq.fasta data_set_fullseq -embedder pt --gpu -bs 0
```

When run as above, everything works fine. However, if I use the `-nproc 2` flag, the computations are much faster, but the resulting database file contains embeddings for only half of the input sequences.

**Comment by Argusmocny** [comment]
`-nproc` will only work correctly when `asdir` or `h5py` option is specified, this should be validated before actual run

**Comment by jludwiczak** [comment]
I've stumbled upon this too - there are also 2 overlapping progress bars shown as I recall.
Additionally there's a typo in README that I've followed at that time: https://github.com/labstructbioinf/pLM-BLAST/blob/main/README.md?plain=1#L64

**Comment by staszekdh** [comment]
Yes, the progress bars looked weird (one was updating during the calculations and the other was stuck at 20%).
